crates/fluvio-connector-package/src/secret.rs

```diff
@@ -453,16 +613,35 @@ config:
         name: secret3
     - secret:
         name: secret4
+  string_interpolation_in_quotes: "My name is ${{ secrets.secret5 }}"
+  string_interpolation_without_quotes: ${{ secrets.secret6 }}
+  string_interpolation_without_spaces: ${{secrets.secret7}}
+  key: value
+  num: 100
+  bool: true
+  obj:
+    key: value
 "#;
 
         //when
         let secrets = detect_secrets_from_str(raw_config).expect("invalid raw config string");
 
         //then
-        assert_eq!(secrets.len(), 4);
+        assert_eq!(secrets.len(), 7);
         assert!(secrets.get("secret1").is_some());
         assert!(secrets.get("secret2").is_some());
         assert!(secrets.get("secret3").is_some());
         assert!(secrets.get("secret4").is_some());
+        assert!(secrets.get("secret5").is_some());
+        assert!(secrets.get("secret6").is_some());
+        assert!(secrets.get("secret7").is_some());
+    }
+
+    #[test]
+    fn test_secret_re() {
+        let regex = get_or_init_secrets_re().expect("invalid regex");
+        assert_eq!(2, regex.captures_len());
+        assert!(regex.is_match("${{ secrets.my_secret_name }}"));
+        assert!(!regex.is_match("${{ env.my_env }}"));
     }
```

Comment:
we need negative test cases as well.  For example what if input like:
```
{{ xxxx
```

What if there are two values? and so forth

Reply:
`{{ xxxx` will just be keep as is, since it will not match the regex. That will just be an string. If there are multiple matches they will all be replaced because we are iterating over all the matches. Will add a test for it.

Reply:
any non conforming syntax should trigger error

Reply:
That would require a more sophisticated lexer. Here my plan was just to support string interpolation for `${{ secrets.something }}`.  I can work on a lexer if that is needed

Reply:
I see that there are several of them, but not sure about one generic that could let me use `${{` instead of `{{`. Checking it now